sasl/digest/src/main/java/org/wildfly/security/sasl/digest/AbstractDigestMechanism.java

```diff
@@ -229,154 +229,153 @@ public byte[] wrap(byte[] outgoing, int offset, int len) throws SaslException {
+            }
 
-        byte[] hmac = new byte[10];
-        System.arraycopy(clearText, clearText.length - 10, hmac, 0, 10);
-
-        byte[] decryptedMessage = null;
-        // strip potential padding
-        if (unwrapCipher.getBlockSize() > 0) {
-            int padSize = clearText[clearText.length - 10 - 1];
-            int decryptedMessageSize = clearText.length - 10;
-            if (padSize < 8) {
-                int i = clearText.length - 10 - 1;
-                while (clearText[i] == padSize) {
-                    i--;
+            byte[] hmac = new byte[10];
+            System.arraycopy(clearText, clearText.length - 10, hmac, 0, 10);
+
+            byte[] decryptedMessage = null;
+            // strip potential padding
+            if (unwrapCipher.getBlockSize() > 0) {
+                int padSize = clearText[clearText.length - 10 - 1];
+                int decryptedMessageSize = clearText.length - 10;
+                if (padSize < 8) {
+                    int i = clearText.length - 10 - 1;
+                    while (clearText[i] == padSize) {
+                        i--;
+                    }
+                    decryptedMessageSize = i + 1;
                 }
-                decryptedMessageSize = i + 1;
+                decryptedMessage = new byte[decryptedMessageSize];
+                System.arraycopy(clearText, 0, decryptedMessage, 0, decryptedMessageSize);
+            } else {
+                decryptedMessage = new byte[clearText.length - 10];
+                System.arraycopy(clearText, 0, decryptedMessage, 0, clearText.length - 10);
             }
-            decryptedMessage = new byte[decryptedMessageSize];
-            System.arraycopy(clearText, 0, decryptedMessage, 0, decryptedMessageSize);
-        } else {
-            decryptedMessage = new byte[clearText.length - 10];
-            System.arraycopy(clearText, 0, decryptedMessage, 0, clearText.length - 10);
-        }
 
-        byte[] expectedHmac = computeHMAC(unwrapHmacKeyIntegrity, extractedSeqNum, hmacMD5, decryptedMessage, 0, decryptedMessage.length);
+            byte[] expectedHmac = computeHMAC(unwrapHmacKeyIntegrity, extractedSeqNum, hmacMD5, decryptedMessage, 0, decryptedMessage.length);
 
-        // check hmac-s
-        if (Arrays2.equals(expectedHmac, 0, hmac, 0, 10) == false) {
-            return NO_BYTES;
+            // check hmac-s
+            if (Arrays2.equals(expectedHmac, 0, hmac, 0, 10) == false) {
+                return NO_BYTES;
+            }
+
+            unwrapSeqNum++; // increment only if MAC is valid
+            return decryptedMessage;
         }
 
-        unwrapSeqNum++; // increment only if MAC is valid
-        return decryptedMessage;
     }
 
+    private static final String CLIENT_MAGIC_INTEGRITY = "Digest session key to client-to-server signing key magic constant";
+    private static final String SERVER_MAGIC_INTEGRITY = "Digest session key to server-to-client signing key magic constant";
```

Comment:
Just a minor comment but style wise, it would be good to move these constants to the top of the class with the other variables (i.e., here https://github.com/wildfly-security/wildfly-elytron/blob/2f2e205de51edc18fbd19b85ebbddc68375b9562/sasl/digest/src/main/java/org/wildfly/security/sasl/digest/AbstractDigestMechanism.java#L75).

Reply:
Hi @fjuma ,Thank you for reviewing the PR.I have updated the PR kindly please review it.